opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp

```diff
@@ -115,6 +121,134 @@ namespace Opm
+                cartesian_to_compressed.insert(std::make_pair(global_cell[i], i));
+            }
+        }
+        else {
+            for (int i = 0; i < number_of_cells; ++i) {
+                cartesian_to_compressed.insert(std::make_pair(i, i));
+            }
+        }
+
+    }
+
+
+    template <class GridT>
+    void SimulatorFullyImplicitBlackoilPolymer<GridT>::
+    computeRepRadiusPerfLength(const Opm::EclipseStateConstPtr eclipseState,
+                               const size_t                    timeStep,
+                               const GridT&                    grid,
+                               std::vector<double>&            wells_rep_radius,
+                               std::vector<double>&            wells_perf_length)
+    {
+
+        int number_of_cells = Opm::UgGridHelpers::numCells(grid);
+        const int* global_cell = Opm::UgGridHelpers::globalCell(grid);
+        const int* cart_dims = Opm::UgGridHelpers::cartDims(grid);
+        auto cell_to_faces = Opm::UgGridHelpers::cell2Faces(grid);
+        auto begin_face_centroids = Opm::UgGridHelpers::beginFaceCentroids(grid);
+
+        if (eclipseState->getSchedule()->numWells() == 0) {
+            OPM_MESSAGE("No wells specified in Schedule section, "
+                        "initializing no wells");
+            return;
+        }
+
+        wells_rep_radius.clear();
+        wells_perf_length.clear();
+
+        std::map<int,int> cartesian_to_compressed;
+
+        setupCompressedToCartesian(global_cell, number_of_cells,
+                                   cartesian_to_compressed);
+
+        ScheduleConstPtr          schedule = eclipseState->getSchedule();
+        std::vector<WellConstPtr> wells    = schedule->getWells(timeStep);
+
+        int well_index = 0;
+
+        for (auto wellIter= wells.begin(); wellIter != wells.end(); ++wellIter) {
+             WellConstPtr well = (*wellIter);
+
+             if (well->getStatus(timeStep) == WellCommon::SHUT) {
+                 continue;
+             }
+             {   // COMPDAT handling
+                 CompletionSetConstPtr completionSet = well->getCompletions(timeStep);
+                 for (size_t c=0; c<completionSet->size(); c++) {
+                     CompletionConstPtr completion = completionSet->get(c);
+                     if (completion->getState() == WellCompletion::OPEN) {
+                         int i = completion->getI();
+                         int j = completion->getJ();
+                         int k = completion->getK();
+
+                         const int* cpgdim = cart_dims;
+                         int cart_grid_indx = i + cpgdim[0]*(j + cpgdim[1]*k);
+                         std::map<int, int>::const_iterator cgit = cartesian_to_compressed.find(cart_grid_indx);
+                         if (cgit == cartesian_to_compressed.end()) {
+                             OPM_THROW(std::runtime_error, "Cell with i,j,k indices " << i << ' ' << j << ' '
```

Comment:
This will fail in the parallel case, see the similar code in the WellsManager for comparison. I must admit that I am considering if it would have been better to add this to the WellsManager after all?